### PR TITLE
multiple captions

### LINKED ISSUE #1572: Captions to support multiple paragraphs
**Is your feature request related to a problem? Please describe.**
Caption blocks can contain multiple paragraph blocks.  Currently we only render the first paragraph.

**Describe the solution you'd like**
We render all the paragraphs

**Describe alternatives you've considered**
N/A

**Testing notes**
Example asset on Test: cm55llxvqkgo
Regression testing that two paragraphs appear and there's no overlap on mobile/smaller viewports as a result. No need for Cypress tests to be added.

Dev insight: Unit tests should be sufficient. We can add a story in Storybook to demonstrate we can have multiple paragraphs for a figure caption.

**Additional context**
Article on Test: https://www.test.bbc.co.uk/news/articles/cm55llxvqkgo
Data on Ares Test: https://ares-broker.test.api.bbci.co.uk/api/article/cm55llxvqkgo

- [x] Initially labelled with ["Refinement needed"](https://github.com/bbc/simorgh/labels/Refinement%20Needed)

### PR DESCRIPTION
**Pull request by twinlensreflex**
Resolves #1572

**Overall change:** Iterates through blocks in the Caption container in order to render more than one caption (should there be more than one).

Screen Shot 2019-04-11 at 13 36 35

**Storybook links:**
`/?selectedKind=Figure&selectedStory=with a caption with multiple paragraphs with a link&full=0&addons=1&stories=1&panelRight=0`

**Code changes:**
`http://localhost:9001/?selectedKind=Figure%20-%20AMP&selectedStory=with%20a%20caption%20with%20multiple%20paragraphs%20with%20a%20link&full=0&addons=1&stories=1&panelRight=0`

- I've simplified `renderText()` so that it only renders text
- I've pulled the `<Caption>` element into its own helper function 
- I did this so that making captions by iterating through blocks is more readable
- I've also added the same article data https://www.test.bbc.co.uk/news/articles/cm55llxvqkgo for local testing

---

- [x] I have assigned myself to this PR and the corresponding issues
- [ ] Tests added for new features
- [ ] Test engineer approval
- [ ] I have followed [the merging checklist](https://github.com/bbc/simorgh-infrastructure/blob/latest/documentation/MERGE_PROCESS.md) and this is ready to merge.

### CODE REVIEW

File: src/app/containers/Caption/index.jsx [hunk 1]
```diff
@@ -9,22 +9,32 @@ import InlineLink from '../InlineLink';
...
-      <Caption>
-        {imageCaptionOffscreenText ? (
-          <VisuallyHiddenText>{imageCaptionOffscreenText}</VisuallyHiddenText>
-        ) : null}
-        {renderText(block)}
-      </Caption>
-    )}
+    {({ imageCaptionOffscreenText }) =>
+      renderMultipleCaptions(
+        block.model.blocks[0].model.blocks,
```
Comment (sareh) [inline src/app/containers/Caption/index.jsx]: There's a [deepGet utility that you can use](https://github.com/bbc/simorgh/blob/latest/src/app/helpers/json/deepGet.js), that is a safer way to access deeply nested items in JSON. 
```suggestion
        deepGet(['model', 'blocks', [0], 'model', 'blocks'], block)
```

File: src/app/containers/Caption/index.jsx [hunk 2]
```diff
@@ -9,22 +9,32 @@ import InlineLink from '../InlineLink';
...
+    {imageCaptionOffscreenText ? (
+      <VisuallyHiddenText>{imageCaptionOffscreenText}</VisuallyHiddenText>
+    ) : null}
+    {renderText(block)}
+  </Caption>
+);
+
+const renderMultipleCaptions = (blocks, imageCaptionOffscreenText) =>
+  blocks.map(block =>
+    renderCaption(block.model.blocks, imageCaptionOffscreenText),
```
Comment (sareh) [inline src/app/containers/Caption/index.jsx]: Likewise you can use the `deepGet` utility here to safely retrieve the nested blocks. `deepGet(['model', 'blocks'], block)`

File: src/app/containers/Caption/index.jsx [hunk 3]
```diff
@@ -9,22 +9,32 @@ import InlineLink from '../InlineLink';
```
Comment (sareh) [inline src/app/containers/Caption/index.jsx:10]: It seems like you haven't pushed your Storybook changes that you had locally to the branch. Can you please push these?

File: data/test/news/articles/cm55llxvqkgo.json [hunk 4]
```diff
@@ -0,0 +1,522 @@
+{
```
Comment (sareh) [inline data/test/news/articles/cm55llxvqkgo.json]: After pushing the Storybook examples, you can remove this extra test article.

**Review by sareh — CHANGES_REQUESTED** [review]
Great work with showing multiple paragraphs in the caption. I've suggested some changes below.

File: src/app/containers/Caption/index.jsx (hunk 1 above)
Comment (dr3) [reply src/app/containers/Caption/index.jsx]: I think its `0` instead of `[0]`, but yay for safe nested objects :D 🎉

**Review by oluoluoxenfree — APPROVED** [review]

**Review by sareh — APPROVED** [review]
👍 Great work!

**Comment by jamesbrumpton** [comment]
The general functionality of this looks to be fine. However, when using VoiceOver each line is treated as a separate caption, which means the user will hear "Caption" every time they go to a new line. Is this expected behaviour?

**Comment by sareh** [comment]
@jamesbrumpton This appears to be the desired functionality, since when going onto the second and subsequent paragraphs in the caption, it clarifies that these are still part of the same caption and that we haven't returned to the article body copy yet.

**Comment by jamesbrumpton** [comment]
Ok. Happy for this to be merged.

**Pull request merged**